pipeline/transform.py

```diff
@@ -19,5 +19,40 @@ def clean_urls(df: pd.DataFrame) -> pd.DataFrame:
     pass
 
 
+def handle_missing_values(df: pd.DataFrame) -> pd.DataFrame:
+    """Handles or removes rows with missing essential fields."""
+    pass
+
+
+def group_sales_by(df: pd.DataFrame, group_by: str = "date") -> pd.DataFrame:
+    """Aggregates total revenue and sales by specified field."""
+    pass
+
+
+def filter_by_currency(df: pd.DataFrame, accepted: list = ["USD", "GBP", "EUR"]) -> pd.DataFrame:
+    """Filters rows based on accepted currency types."""
+    pass
+
+
+def validate_forein_keys(df: pd.DataFrame) -> bool:
+    """Checks if a forein key already exists before mapping it to a new id."""
+    pass
+
+
+def get_id_mapping(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
For this function, would you want to pass in the table name as a parameter? We may not need to return a dataframe either, possibly just a dictionary of keys mapped to names.
```suggestion
def get_id_mapping(df: pd.DataFrame, table_name: str) -> dict:
```

Reply:
I thought I would need to add the ids as columns in the dataframe?

Reply:
I would also like some clarification on this change